lumigator/python/mzai/backend/api/deps.py

```diff
@@ -34,9 +35,19 @@ def get_s3_client() -> Generator[S3Client, None, None]:
 S3ClientDep = Annotated[S3Client, Depends(get_s3_client)]
 
 
-def get_dataset_service(session: DBSessionDep, s3_client: S3ClientDep) -> DatasetService:
+def get_s3_filesystem() -> Generator[S3FileSystem, None, None]:
```

Comment:
I guess we'll end up with file-like objects regardless of where they're placed, either in local FS or in S3, right?

Reply:
Yes, file-like or directory-like (as in the case of HF datasets).
I realise there is a bit of redundancy with both `s3_client` and `S3FileSystem`: these are used together to implement different s3-based tasks but I am quite sure we could consolidate them into just one if we choose to continue writing our own data storage system